tests/test_api.py

```diff
@@ -212,6 +223,21 @@ def test_sign_verify(self) -> None:
         with self.assertRaises(exceptions.UnsignedMetadataError):
             targets_key.verify_signature(md_obj)
 
+    def test_sign_failures(self) -> None:
+        # Make sure there is a case where we throw UnsignedMetadataError.
+        # If one day securesystemslib change their implementation and throw
+        # something different than UnsignedMetadataError,
+        # then make sure the metadata API sign doc will change too.
```

Comment:
this comment is incorrect (sslib does not raise UnsignedMetadataError) and very long

Reply:
Ohhh... such a bad comment. :D 
Yea, I am definitely updating it.